File: NameSiloDynDns/HostToUpdate.cs
using System;

namespace NameSiloDynDns
{
    public class HostToUpdate
    {
        public string Host { get; private set; }
        public string Domain { get; set; }
        public int Hours { get; private set; }
        public int Minutes { get; private set; }
        public int Seconds { get; private set; }
        public int RetryAttempts { get; private set; }
        public TimeSpan UpdateTimeSpan => new TimeSpan(Hours, Minutes, Seconds);
        public TimeSpan RetryTimeSpan =>
            RetryAttempts < 0
            ? TimeSpan.Zero
            : TimeSpan.FromTicks(UpdateTimeSpan.Ticks / (RetryAttempts + 1));
    }
}

File: NameSiloDynDns/HostToUpdate.fs
namespace NameSiloDynDns

open System

[<CLIMutable>]
type HostToUpdate =
    { Host: string
      Domain: string
      Hours: int
      Minutes: int
      Seconds: int
      RetryAttempts: int }

    member this.UpdateTimeSpan =
        TimeSpan(this.Hours, this.Minutes, this.Seconds)

    member this.RetryTimeSpan =
        if this.RetryAttempts < 0 then
            TimeSpan.Zero
        else
            TimeSpan.FromTicks(
                this.UpdateTimeSpan.Ticks
                / int64 (this.RetryAttempts + 1)
            )
